app/assets/javascripts/discourse/app/lib/ajax.js

```diff
@@ -29,14 +29,9 @@ export function handleLogoff(xhr) {
   }
 }
 
-function handleRedirect(data) {
-  if (
-    data &&
-    data.getResponseHeader &&
-    data.getResponseHeader("Discourse-Xhr-Redirect")
-  ) {
-    window.location.replace(data.responseText);
-    window.location.reload();
+function handleRedirect(xhr) {
+  if (xhr && xhr.getResponseHeader("Discourse-Xhr-Redirect")) {
+    window.location = xhr.responseText;
```

Comment:
Are we sure the behaviour of just changing `location` is the same as using `replace` and `reload`?

Maybe it was done for some weird browser / os combination?

Reply:
I am not sure if it was done for a browser / OS combination, but from my tests it did not work anywhere. It simply refreshed the current page.

Reply:
Looks like it was done 4 years ago, maybe the API changed since? 🤔 

@SamSaffron any ideas? Looks like you were the one who added this `handleRedirect` method.